File: Chapter3/jello/MazeBuilder.py
import abc

# Builder
class MazeBuilder:
	__metaclass__ = abc.ABCMeta

	def build_maze(self):
		pass

	def build_room(self):
		pass

# A derived class from Builder
class StandardMazeBuilder(MazeBuilder):
	def build_maze(self):
		# logic about enchanted maze
		pass

	def build_room(self):
		# logic about enchanted room
		pass



# use it
def create_maze(maze_bulider)
	maze = maze_bulider.build_maze()
	room = maze_bulider.build_room()

if __name__ == "__main__":
	maze_bulider = EnchantedMazeFactory()
    create_maze(maze_bulider)

File: Chapter3/demarlik01/playground/src/com/demarlik01/patterns/builder/MazeBuilder.java
package com.demarlik01.patterns.builder;

import com.demarlik01.materials.Maze;

public class MazeBuilder {
  public void buildMaze(){
  }

  public void buildRoom(Integer room){
  }

  public void buildDoor(Integer roomFrom, Integer roomTo){
  }

  protected MazeBuilder(){
  }

  public Maze getMaze(){
    return new Maze();
  }
}
